near-sdk/src/store/tree_map/iter.rs

```diff
@@ -584,7 +398,11 @@ where
 
     fn size_hint(&self) -> (usize, Option<usize>) {
         let len = self.length as usize;
-        (0, Some(len))
+        (len, Some(len))
```

Comment:
So the main concern with this is that this is stating that there is an exact amount of elements in this iterator, which is not the case for `KeysRange`. I do think that this difference can be useful to users, if they want to use `Keys` as an exact size iterator.

I would recommend trying to keep `Keys` and `KeysRange` as separate types, but just share the internal logic with shared types or abstracting the methods to just take the raw components.

Reply:
Cool! I'll add `Keys` back and have shared methods.